check-fact-gather-results.sh

```diff
@@ -0,0 +1,163 @@
+        if [ -n "${ROLE_DEFAULT_GATHERS_FACTS[$role]:-}" ]; then
+            match="$default_match"
+        else
+            match="ok:"
+        fi
+    elif [ "$role" = rhc ]; then  # runs some tests with gather_facts: false
+        if [[ "$log_link" =~ tests_register_unregister ]]; then
+            match="ok:"
+        elif [[ "$log_link" =~ tests_repositories ]]; then
+            match="ok:"
+        else
+            match="$default_match"
+        fi
+    elif [ "$role" = postgresql ]; then
+        # requires an unusual fact - will almost always have to gather
+        match="ok:"
+    elif [ "$role" = ad_integration ]; then  # runs a test with gather_facts: false
+        if [[ "$log_link" =~ tests_basic_join ]]; then
+            match="ok:"
+        else
+            match="$default_match"
+        fi
+    elif [ "$role" = kdump ]; then
+        if [ "${ANSIBLE_GATHERING:-no}" = explicit ] && [[ "$log_link" =~ tests_ssh.yml ]]; then
+            match="skipping:"
+        else
+            match="$default_match"
+        fi
+    else
+        match="$default_match"
+    fi
+    if [ "$role" = network ]; then
+        task_match="TASK .*network : Ensure ansible_facts used by role are present"
+    else
+        task_match="TASK .*$role : Ensure ansible_facts used by role"
+    fi
+    curl -s -L "$log_link" | \
+    while read -r line; do
+        # use role to avoid issues with nested roles
+        if [ "${in_task:-false}" = true ]; then
+            if [[ "$line" =~ ^included: ]]; then
+                # this is not the right task
+                in_task=false
+                task_lines=()
+                continue
+            elif [ -z "$line" ]; then
+                echo ERROR - match not found in task
+                printf '%s\n' "${task_lines[@]}"
+                return 1
+            elif [[ "$line" =~ ^${match} ]]; then
+                # success
+                in_task=false
+                task_lines=()
+                # if we already gathered facts in this test, we don't
+                # need to gather again, so subsequent tasks should
+                # report skipping
+                match="skipping:"
+                continue
+            fi
+            task_lines+=("$line")
+        elif [[ "$line" =~ ^${task_match} ]]; then
+            in_task=true
+            task_lines=("$line")
+        elif [[ "$line" =~ ^TASK\ \[Gathering\ Facts\] ]]; then
+            if [ "$role" != postgresql ]; then
+                match="skipping:"
+            fi
+        elif [[ "$line" =~ ^TASK\ \[.*Gather\ the\ minimum\ subset\ of\ ansible_facts ]]; then
+            match="skipping:"
+        elif [[ "$line" =~ ^TASK\ \[.*Ensure\ facts\ used\ by\ test ]]; then
+            match="skipping:"
+        elif [[ "$line" =~ ^TASK\ \[.*Ensure\ Ansible\ facts\ required\ by\ tests ]]; then
+            match="skipping:"
+        fi
+    done
+}
+
+get_logs_from_pr() {
+    local pr role
```

Comment:
```suggestion
    local role pr
```
Only one nitpick, preserve the order of how the function assigns vars